src/component/handlers/edit/editOnInput.js

```diff
@@ -39,14 +41,30 @@ var DOUBLE_NEWLINE = '\n\n';
  * due to a spellcheck change, and we can incorporate it into our model.
  */
 function editOnInput(editor: DraftEditor): void {
+  editOnSelect(editor);
```

Comment:
We should add a comment to the effect of "Ensure the selection is updated as it may have changed since `editOnBeforeInput` was called", maybe with a little bit of detail that there is a window between `onBeforeInput` and `onInput` where the selection may change.